### DISCUSSION TITLE
Add thread wrapper

### DISCUSSION THREAD

**Discussion by SuperDuperDeou** [qtile/qtile#4186] [Ideas]
If I add a thread to the config.py, when the config is reloaded the thread that's already open is not killed, and another instance is open while reloading the config, which ends up with two instances of the same thread running.

So, if I decide to use a thread, I have to make sure it gets killed the very moment the config is reloaded.

I think Qtile could use a wrapper for pthreads that takes care of that.

**Comment by Pakrohk** [comment]
I feel that if you do the following, your problem will be solved.
```py
def main(qtile):
    qtile.configured = False
    ....
    ....
    qtile.configured = True
```

**Comment by m-col** [comment]
You could also use the `startup_once` hook which is only executed when qtile starts up the very first time, regardless of any config reloads or restarts.
